raft.c

```diff
@@ -134,13 +134,32 @@ static void executeRaftRedisCommandArray(RaftRedisCommandArray *array,
         enterRedisModuleCall();
         RedisModuleCallReply *reply = RedisModule_Call(
                 ctx, cmd, "v", &c->argv[1], c->argc - 1);
+        int ret_errno = errno;
         exitRedisModuleCall();
 
         if (reply_ctx) {
             if (reply) {
                 RedisModule_ReplyWithCallReply(reply_ctx, reply);
             } else {
-                RedisModule_ReplyWithError(reply_ctx, "ERR Unknown command or wrong number of arguments");
+                /* Try to produce an error message which is similar to Redis */
+                int trunc_cmdlen = cmdlen > 256 ? 256 : cmdlen;
+                size_t errmsg_len = 128 + trunc_cmdlen;   /* Big enough for msg + cmd */
+                char *errmsg = RedisModule_Alloc(errmsg_len);
+
+                switch (ret_errno) {
+                    case ENOENT:
+                        snprintf(errmsg, errmsg_len, "ERR unknown command `%.*s`", trunc_cmdlen, cmd);
+                        break;
+                    case EINVAL:
+                        snprintf(errmsg, errmsg_len, "ERR wrong number of arguments for '%.*s' command",
+                                trunc_cmdlen, cmd);
+                        break;
+                    default:
+                        snprintf(errmsg, errmsg_len, "ERR failed to execute command '%.*s'",
+                                trunc_cmdlen, cmd);
+                }
+                RedisModule_ReplyWithError(reply_ctx, errmsg);
+                RedisModule_Free(errmsg);
```

Comment:
this seems reasonable, but should redis itself have a sprintf style ReplyWithErrorf() function?

Reply:
Yes it could, or alternatively just have `RedisModule_ReplyWithErrorString` that takes a `RedisModuleString`. But I don't think it really matters.